src/libAtomVM/mailbox.c

```diff
@@ -44,10 +44,12 @@ void mailbox_init(Mailbox *mbx)
 // not call it. They should call mailbox_remove instead.
 void mailbox_destroy_signal_message(MailboxMessage *m)
 {
-    if (m->header.type == NormalMessage) {
-        memory_sweep_mso_list(m->body.normal.mso_list);
-    } else if (m->header.type == KillSignal || m->header.type == TrapAnswerSignal) {
-        memory_sweep_mso_list(m->body.term.mso_list);
+    if (m->type == NormalMessage) {
+        Message *normal_message = CONTAINER_OF(m, struct Message, base);
```

Comment:
We have an inconsistent use of `Message` and `struct Message` here, haven't we?
I guess `struct should` be internal and `typedef` should be exported. So maybe `MailboxMessage` should only be a struct now, as it's the old `struct MessageHeader`.

Reply:
I agree with the general rule. I changed all `struct Message` to just `Message`.